definitions/marts/looker_studio_marts/ls_declarations_provider_names.sqlx

```diff
@@ -40,17 +42,27 @@ config {
 
 SELECT
   dec.*,
+  COALESCE(ecfid.cohort,npqe.cohort) AS cohort,
+  COALESCE(ecfid.schedule_identifier,npqe.schedule_identifier) AS schedule_identifier,
   COALESCE(lpl.name,npqlpl.name) AS cpd_lp_name,
   dpl.name AS cpd_dp_name,
   CASE
-    WHEN course_identifier LIKE 'npq%' THEN 'NPQ'
-    WHEN course_identifier LIKE 'ecf%' THEN 'ECF'
+    WHEN dec.course_identifier LIKE 'npq%' THEN 'NPQ'
+    WHEN dec.course_identifier LIKE 'ecf%' THEN 'ECF'
   ELSE
   'Other'
 END
   AS programme
 FROM
   ${ref("participant_declarations_latest_cpd")} dec
+LEFT JOIN
+  ${ref("ecf_inductions_dedupe")} ecfid
+USING
+  (participant_profile_id)
+LEFT JOIN
+  ${ref("npq_enrolments")} npqe
+USING

```

Comment:
Could you switch the join here to join the `participant_profile_id` from `declarations_latest` with the `application_ecf_id` in `npq_enrolments`? That way, if we change the naming of the ambiguous fields in npq_enrolments, it won't affect this mart.

Reply:
Have changed!